tests/subsys/tracing/tracing_api/src/main.c

```diff
@@ -0,0 +1,224 @@
+#define STACK_SIZE (1024 + CONFIG_TEST_EXTRA_STACKSIZE)
+
+/* Check flags */
+bool thread_api_found;
+bool isr_api_found;
+bool condvar_api_found;
+bool sem_api_found;
+bool mutex_api_found;
+bool timer_api_found;
+bool data_format_found;
+bool raw_data_format_found;
+bool sync_string_format_found;
+
+#if defined(CONFIG_TRACING_BACKEND_UART)
+static void tracing_backends_output(
+		const struct tracing_backend *backend,
+		uint8_t *data, uint32_t length)
+{
+	/* Check the output data. */
+	if (strstr(data, "sys_trace_k_thread") != NULL) {
+		thread_api_found = true;
+	}
+	if (strstr(data, "sys_trace_isr") != NULL) {
+		isr_api_found = true;
+	}
+	if (strstr(data, "sys_trace_k_condvar") != NULL) {
+		condvar_api_found = true;
+	}
+	if (strstr(data, "sys_trace_k_sem") != NULL) {
+		sem_api_found = true;
+	}
+	if (strstr(data, "sys_trace_k_mutex") != NULL) {
+		mutex_api_found = true;
+	}
+	if (strstr(data, "sys_trace_k_timer") != NULL) {
+		timer_api_found = true;
+	}
+	if (strstr(data, "tracing_format_data_testing") != NULL) {
+		data_format_found = true;
+	}
+	if (strstr(data, "tracing_format_raw_data_testing") != NULL) {
+		raw_data_format_found = true;
+	}
+	if (strstr(data, "tracing_format_string_testing") != NULL) {
+		sync_string_format_found = true;
+	}
+}
+#endif
+
+#if defined(CONFIG_TRACING_BACKEND_UART)
+const struct tracing_backend_api tracing_uart_backend_api = {
+	.init = NULL,
+	.output  = tracing_backends_output
+};
+
+TRACING_BACKEND_DEFINE(tracing_backend_uart, tracing_uart_backend_api);
+#endif
+
+#ifdef CONFIG_TRACING_ASYNC
+void test_tracing_sys_api(void)
+{
+	int ret = 0, prio = 0;
+	size_t stack = 0;
+	struct k_mutex mutex;
+	struct k_thread thread;
+	struct k_condvar condvar;
+	struct k_sem sem, sem2;
+	struct k_timer timer;
+	k_timeout_t timeout = K_MSEC(1);
+
+	tracing_buffer_init();
+	/* thread api */
+	sys_trace_k_thread_switched_out();
+	sys_trace_k_thread_switched_in();
+	sys_trace_k_thread_priority_set(&thread);
+	sys_trace_k_thread_sched_set_priority(&thread, prio);
+	sys_trace_k_thread_create(&thread, stack, prio);
+	sys_trace_k_thread_start(&thread);
+	sys_trace_k_thread_abort(&thread);
+	sys_trace_k_thread_suspend(&thread);
+	sys_trace_k_thread_resume(&thread);
+	sys_trace_k_thread_ready(&thread);
+	sys_trace_k_thread_sched_ready(&thread);
+	sys_trace_k_thread_sched_abort(&thread);
+	sys_trace_k_thread_sched_resume(&thread);
+	sys_trace_k_thread_sched_suspend(&thread);
+	sys_trace_k_thread_sleep_enter(timeout);
+	sys_trace_k_thread_sleep_exit(timeout, ret);
+	sys_trace_k_thread_abort_enter(&thread);
+	sys_trace_k_thread_abort_exit(&thread);
+	sys_trace_k_thread_yield();
+	sys_trace_k_thread_wakeup(&thread);
+	sys_trace_k_thread_pend(&thread);
+	sys_trace_k_thread_info(&thread);
+	sys_trace_k_thread_name_set(&thread, ret);
+	sys_trace_k_thread_sched_lock();
+	sys_port_trace_k_thread_sched_unlock();
+	sys_trace_k_thread_join_blocking(&thread, timeout);
+	sys_trace_k_thread_join_exit(&thread, timeout, ret);
```

Comment:
It seems to me that "thread_api_found" will be set if any of the sys_trace_k_thread_* is called successfully. I don't think it is correct.

Reply:
Hi @LeiW000 , fixed, could you like to review this, thanks.